app/channels/subscribe_channel.rb

```diff
@@ -0,0 +1,40 @@
+class SubscribeChannel < BaseSharingChannel
+  def subscribed
+    Rails.logger.info("[SubscribeChannel#subscibed] #{params.inspect}")
+    @subscriber_channel_id = params["channelId"]
+    return unless validate_subscriber_channel_id_format
+    @publisher_channel_id = Rails.cache.fetch(["subscriber_to_publisher", @subscriber_channel_id])
+
+    if @publisher_channel_id
+      if metadata[:last_used_at] < 2.hours.ago
+        transmit(format_error("Old Channel"))
+        return
+      else
+        transmit(metadata[:last_message])
+      end
+    else
+      transmit(format_error("Channel Not Found"))
+      return
+    end
+
+    stream_for @subscriber_channel_id
+  end
+
+  def unsubscribed
+  end
+
+  protected
+
+  def validate_subscriber_channel_id_format
```

Comment:
This is quite similar to `validate_publisher_channel_id_format` - perhaps both could be factored out into the superclass?

Reply:
I was thinking since the publisher ids come from @frett's code, but subscriber ids come from our code, we may want different regex's if they're different, but if not I'm good to combine this

Reply:
or maybe if that is the case I can just add a block for regex

Reply:
We ought to be able to agree on reasonable and identical regex for both.

Reply:
honestly I don't care too much about the ids following a specific format beyond these requirements:
- needs to be URI safe because the value is passed to receivers via a url param
- needs to have enough entropy to avoid collisions and prevent guessing of the values